src/pages/groupComparison/ComparisonGroupManager.tsx

```diff
@@ -0,0 +1,285 @@
+import * as React from "react";
+import {observer} from "mobx-react";
+import {StudyViewPageStore, UniqueKey} from "../studyView/StudyViewPageStore";
+import {
+    addGroupToLocalStorage, addSamplesToGroup,
+    deleteGroups,
+    getLocalStorageGroups,
+    restoreRecentlyDeletedGroups
+} from "./GroupPersistenceUtils";
+import {computed, observable} from "mobx";
+import autobind from "autobind-decorator";
+import {SyntheticEvent} from "react";
+import {getDefaultGroupName} from "./GroupComparisonUtils";
+import _ from "lodash";
+import {SampleIdentifier} from "../../shared/api/generated/CBioPortalAPI";
+import {getComparisonUrl} from "../../shared/api/urls";
+import "./styles.scss";
+import ReactSelect from "react-select";
+
+export interface IComparisonGroupManagerProps {
+    store:StudyViewPageStore;
+}
+
+@observer
+export default class ComparisonGroupManager extends React.Component<IComparisonGroupManagerProps, {}> {
+    @observable groupNameFilter:string = "";
+    @observable recentlyDeleted = false;
+    @observable addGroupPanelOpen = false;
+    @observable _inputGroupName:string = "";
+    @computed get inputGroupName() {
+        return this._inputGroupName;
+    }
+    @observable addSamplesTargetGroupId:string = "";
+
+    @autobind
+    private onChangeGroupNameFilter(e:SyntheticEvent<HTMLInputElement>) {
+        this.groupNameFilter = (e.target as HTMLInputElement).value;
+    }
+
+    @autobind
+    private onChangeInputGroupName(e:SyntheticEvent<HTMLInputElement>) {
+        this._inputGroupName = (e.target as HTMLInputElement).value;
+    }
+
+    @computed get filteredGroups() {
+        // TODO: fuzzy string search?
+        return this.props.store.comparisonGroups.filter(group=>group.name.toLowerCase().indexOf(this.groupNameFilter.toLowerCase()) > -1);
+    }
+
+    @autobind
+    private showAddGroupPanel() {
+        this.addGroupPanelOpen = true;
+    }
+
+    @autobind
+    private cancelAddGroup() {
+        this.addGroupPanelOpen = false;
+        this._inputGroupName = "";
+        this.addSamplesTargetGroupId = "";
+    }
+
+    private get header() {
+        return (
+            <div>
+                <h5 style={{position:"absolute", top:7}}>Groups</h5>
```

Comment:
curious why h5 is position absolute.  If it's just to bump it down, better to do with margin-top so it stays in flow.